python/tidy/linting_report.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2025 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from typing import List, Literal, NotRequired, TypedDict
+
+
+class GithubAnnotation(TypedDict):
+    file_name: str
+    line_start: int
+    line_end: int
+    level: Literal["notice", "warning", "error"]
+    title: str
+    message: str
+    column_start: NotRequired[int]
+    column_end: NotRequired[int]
+
+
+class LintingReportManager:
```

Comment:
Let's call this `GitHubAnnotationManager` as it might be used in more cases than just linting in the future. Then the file would be `github_annotations.py` or something similar.